### ISSUE TITLE
Create instasearch.js adaptor

### ISSUE THREAD

**Issue by windbeneathyourwings** [opensearch-project/OpenSearch#2072]
Hello,

I think it would be pretty cool to be able to use Agnolia Instasearch.js with Open Search. The only low cost alternative that exist at the moment is typesense. The typesense adaptor could be used as reference to create the open search adaptor. 

Agnolia Instasearch

https://www.algolia.com/doc/

Typesense adaptor

https://github.com/typesense/typesense-instantsearch-adapter

**Comment by CEHENKLE** [comment]
Hey @windbeneathyourwings !  Thanks for the note.  Is this something you'd be interested in contributing?  It does sound like a cool idea.

**Comment by windbeneathyourwings** [comment]
Hello @CEHENKLE 

I'm considering it because I would like to provide it as part of my own project which exclusively uses open search for search. Its just a matter of prioritizing it considering I work on all this stuff outside of work.

https://github.com/ng-druid/platform

**Comment by ahopp** [comment]
@windbeneathyourwings I haven't seen anyone working on anything similar yet. 

My understanding is that you'd want to build an adapter so you'd be able to use [Instantsearch](https://github.com/algolia/instantsearch.js) (and potentially related [React](https://github.com/algolia/react-instantsearch), [Vue](https://github.com/algolia/vue-instantsearch) and [Angular](https://github.com/algolia/angular-instantsearch) tools) with data indexed in OpenSearch (as opposed to a Typesense search server). Is that correct?

**Comment by windbeneathyourwings** [comment]
@ahopp 

Yes, create an adaptor like the type sense one but for open search.

**Comment by minalsha** [comment]
@windbeneathyourwings  are you still working on contributing to this issue? Thanks

**Comment by windbeneathyourwings** [comment]
@minalsha Yes, I still plan on building an open search adaptor unless someone takes it on before able to prioritize.

**Comment by cfrancoi** [comment]
Hi !
i think this project for ES / OS with a instant search adapter : [search](https://github.com/searchkit/searchkit) do the work if some people can confirm.

**Comment by ahopp** [comment]
Looks like some of it is covered, e.g.,

> Searchkit provides a Search UI for Elasticsearch or Opensearch. With Searchkit, you can use Instantsearch components like Searchbox, refinement filters and results (and many more!) to build a search experience.

But it might be worthwhile from some validation since it still explicitly calls out an ES specification (e.g., Searchkit Node API proxies Elasticsearch requests from the browser and the use of Elasticsearch Query DSL for advanced queries) despite usability with OpenSearch (specifically, Opensearch 2.4 or higher).
